In [1]:
import pandas as pd
from neuronunit.optimization.optimization_management import public_opt
from sciunit.scores import RelativeDifferenceScore, ZScore
from sciunit.utils import config_set, config_get
config_set('PREVALIDATE', False)
from neuronunit.allenapi.allen_data_driven import make_allen_hard_coded_limited, make_allen_hard_coded_complete
import warnings
from neuronunit.optimization.model_parameters import BPO_PARAMS
warnings.filterwarnings("ignore")
[allen_suite_623960880,_,_,_,_,df] = make_allen_hard_coded_complete()

/home/user/anaconda3/lib/python3.7/site-packages/cerberus/validator.py:1609: UserWarning: No validation schema is defined for the arguments of rule 'not_zero_obs_zscore'
  "'%s'" % method_name.split('_', 2)[-1]


# In this notebook
we fit Izhikevich and Adaptive Exponential Integrate and fire models to electrophysiology observations 
from the Allen Brain, using BluePyOpt and NeuronUnit.

* Immediately Below 
is a list of some of the observations we will use:

In [2]:
df

TimeConstantTest       name RestingPotentialTest  \
623960880          23.8 ms  623960880             -65.1 mV   
623893177          27.8 ms  623893177             -77.0 mV   
471819401          13.8 ms  471819401             -77.5 mV   
482493761          24.4 ms  482493761             -71.6 mV   

          InputResistanceTest RheobaseTest  
623960880       241.0 megaohm      70.0 pA  
623893177       136.0 megaohm     190.0 pA  
471819401       132.0 megaohm     190.0 pA  
482493761       132.0 megaohm      70.0 pA

Allen Specimen ID: 471819401
Four data fitting constraints:
* RheobaseTest	
* TimeConstantTest	
* RestingPotentialTest	
* InputResistanceTest	
# Model AdExp

In [ ]:
_,_,_,a_cells = make_allen_hard_coded_limited()
opt,obs_preds,df = public_opt(
    a_cells['471819401'],
    BPO_PARAMS,
    '471819401',
    "ADEXP",
    100,
    300,
    "IBEA",
    score_type=RelativeDifferenceScore
    )


In [ ]:
df

# Solution Model Parameters:


In [ ]:
opt.attrs_to_params()
opt.params

Allen Specimen ID: 471819401
Four data fitting constraints:
* RheobaseTest	
* TimeConstantTest	
* RestingPotentialTest	
* InputResistanceTest	
# Model Izhikevich

In [ ]:
opt,obs_preds,df = public_opt(
    a_cells['471819401'],
    BPO_PARAMS,
    'fi_curve',
    "IZHI",
    100,
    360,
    "IBEA",
    )

In [ ]:
df

# Solution Parameters:

In [ ]:
opt.attrs_to_params()
opt.params

# Just FI curve on AdExp model
Allen Specimin Id: 471819401

In [ ]:
opt,obs_preds,df = public_opt(
    a_cells['fi_curve'],
    BPO_PARAMS,
    'fi_curve',
    "ADEXP",
    100,
    100,
    "IBEA"
    )

In [ ]:
df


In [ ]:
opt.attrs

Allen Specimen ID: 623960880
Four data fitting constraints:
* RheobaseTest	
* TimeConstantTest	
* RestingPotentialTest	
* InputResistanceTest	
# Model Izhikevich

In [ ]:

opt,obs_preds,df = public_opt(
    allen_suite_623960880.tests,
    BPO_PARAMS,
    allen_suite_623960880.name,
    "IZHI",
    100,
    100,
    "IBEA"
    )

In [ ]:
df

# Solution parameters

In [ ]:
opt.attrs_to_params()
opt.params